src/daemon.rs

```diff
@@ -371,8 +375,7 @@ impl Daemon {
     }
 
     pub fn list_blk_files(&self) -> Result<Vec<PathBuf>> {
-        let mut path = self.daemon_dir.clone();
-        path.push("blocks");
+        let mut path = self.blocks_dir.clone();
         path.push("blk*.dat");
```

Comment:
I'd suggest changing these two lines to `let path = self.blocks_dir.join("blk*.dat");` it's more obvious and more efficient.